agent/local/state.go

```diff
@@ -590,6 +596,13 @@ func (l *State) UpdateCheck(id types.CheckID, status, output string) {
 		return
 	}
 
+	// Ensure we only mutate a copy of the check state and put the finalized
+	// version into the checks map when complete.
+	c = c.Clone()
+	defer func(c *CheckState) {
+		l.checks[id] = c
+	}(c)
```

Comment:
We are relying on the defer to run before the deferred unlock. That's fine per Go's spec, but I wonder if we should make it more obvious in a comment here or at the top that if someone ever removes or moves the defer unlock it would be racy?

Reply:
Left a note.